core/trino-main/src/main/java/io/trino/execution/executor/MultilevelSplitQueue.java

```diff
@@ -304,6 +313,19 @@ public static int computeLevel(long threadUsageNanos)
         return LEVEL_THRESHOLD_SECONDS.length - 1;
     }
 
+    public DistributionStat getWaitingSplitsSizeDistribution()
```

Comment:
I see that `waitingSplits.size() `is already exposed. Why do we need `DistributionStat`?

Reply:
`waitingSplits.size` gives you only *current* size of the `waitingSplits` queue. As our last investigation showed, it could be very helpful to be able to answer the question: What is the size of the queue by 90% percentage of time within the last 5/10/15 minute period? If you don't collect the distribution weighted in the time you just have only `waitingSplits.size()` in any point in time.